samples/dnn/dasiamrpn_tracker.cpp

```diff
@@ -0,0 +1,468 @@
+        int hs = int(ws * ratios[i]);
+
+        float wws = float(ws)*scales;
+        float hhs = float(hs)*scales;
+        Rect2f anchor = {0,0,wws,hhs};
+        baseAnchors.push_back(anchor);
+    }
+
+    int anchorIndex[] = {0,0,0,0};
+    const int sizes[] = {4, (int)ratios.size(), scoreSize, scoreSize};
+    Mat anchors(4, sizes, CV_32F);
+
+    for(auto i = 0; i< scoreSize; i++)
+    {
+        for(auto j = 0; j < scoreSize; j++)
+        {
+            for(auto k = 0; k < anchorNum; k++)
+            {
+                anchorIndex[0] = 1, anchorIndex[1] = k, anchorIndex[2] = i, anchorIndex[3] = j;
+                anchors.at<float>(anchorIndex) = ori + totalStride*i;
+
+                anchorIndex[0] = 0;
+                anchors.at<float>(anchorIndex) = ori + totalStride*j;
+
+                anchorIndex[0] = 2;
+                anchors.at<float>(anchorIndex) = baseAnchors[k].width;
+
+                anchorIndex[0] = 3;
+                anchors.at<float>(anchorIndex) = baseAnchors[k].height;
+            }
+        }
+    }
+
+    return anchors;
+}
+
+Mat getSubwindow(Mat &img, const Rect2f &targetBox, float originalSize, Scalar avgChans)
+{
+    Mat zCrop;
+    Size imgSize = img.size();
+    float c = (originalSize+1)/2;
+    float xMin = cvRound(targetBox.x - c);
+    float xMax = xMin + originalSize - 1;
+    float yMin = cvRound(targetBox.y - c);
+    float yMax = yMin + originalSize - 1;
+
+    int leftPad = (int)(fmax(0., -xMin));
+    int topPad = (int)(fmax(0., -yMin));
+    int rightPad = (int)(fmax(0., xMax - imgSize.width +1));
+    int bottomPad = (int)(fmax(0., yMax - imgSize.height +1));
+
+    xMin = xMin + leftPad;
+    xMax = xMax + leftPad;
+    yMax = yMax + topPad;
+    yMin = yMin + topPad;
+
+    if(topPad == 0 && bottomPad ==0 && leftPad == 0 && rightPad == 0)
+    {
+        img(Rect(int(xMin), int(yMin), int(xMax - xMin + 1), int(yMax - yMin + 1))).copyTo(zCrop);
+    } else
+    {
+        copyMakeBorder(img, img, topPad, bottomPad, leftPad, rightPad, BORDER_CONSTANT, avgChans );
+        img(Rect(int(xMin), int(yMin), int(xMax - xMin + 1), int(yMax - yMin + 1))).copyTo(zCrop);
+    }
+
+    return zCrop;
+}
+
+void softmax(const Mat &src, Mat & dst)
+{
+    Mat maxVal;
+    cv::max(src.row(1), src.row(0), maxVal);
+
+    src.row(1) -= maxVal;
+    src.row(0) -= maxVal;
+
+    exp(src, dst);
+
+    Mat sumVal = dst.row(0) + dst.row(1);
+    dst.row(0) = dst.row(0)/sumVal;
+    dst.row(1) = dst.row(1)/sumVal;
+
+}
+
+void elementMax(Mat & src)
+{
+    int *p = src.size.p;
+    int index[] = {0,0,0,0};
+    for(int n = 0; n< *p; n++)
+    {
+        for(int k = 0; k< *(p+1) ; k++)
+        {
+            for(int i = 0; i < *(p+2); i++)
+            {
+                for(int j = 0; j < *(p+3); j++)
+                {
+                    index[0] = n, index[1] = k, index[2] = i, index[3] = j;
+                    src.at<float>(index) = fmax(src.at<float>(index), (float)(1.0/src.at<float>(index)));
```

Comment:
`.at()` is evaluated 3 times here.

Use this code instead:

```
float& v = src.at<float>(index);
v = fmax(v, 1.0f / v);
```

Reply:
Thank you for your careful review. I have used your comment in new version.